src/python/pants/backend/python/tasks/select_interpreter.py

```diff
@@ -65,15 +66,17 @@ def execute(self):
     if not python_tgts_and_reqs:
       return
     python_tgts = [tgt for tgt in python_tgts_and_reqs if isinstance(tgt, PythonTarget)]
-    fs = PythonInterpreterFingerprintStrategy()
+    interpreter_cache = PythonInterpreterCache.global_instance()
+    fs = PythonInterpreterFingerprintStrategy(python_setup=interpreter_cache._python_setup)
```

Comment:
I think this is right to refer to the interpreter cache's value of `PythonSetup`, rather than this file making its own explicit link to it. 

But this is a private member, so not sure?

Reply:
If we wanted to do this, I would probably create a `PythonInterpreterFingerprintStrategy` in `PythonInterpreterCache` and allow for just:
```python
    fs = PythonInterpreterCache.global_instance().fingerprint_strategy
```
or it could be a `.get_fingerprint_strategy()` method instead of a property.

Reply:
I know we agreed that it would probably be best to have `PythonInterpreterCache` expose `._python_setup`, but after seeing `DependencyContext` in the jvm backend, I'm thinking that it might be more idiomatic to have the `PythonInterpreterCache` subsystem produce the fingerprint strategy directly: https://github.com/pantsbuild/pants/blob/36fdbd8973763e8b2dae994b25dd19f79cf20301/src/python/pants/backend/jvm/subsystems/dependency_context.py#L50-L51

Reply:
This strategy is (currently) very specific to this task and to selecting a global interpreter for a context. In particular, it doesn't fingerprint anything else about the target: only its constraints. `ResolvedJarAwareFingerprintStrategy` falls back to the target fingerprint, so is a bit more general.

_Should_ it be fingerprinting more of the target so that it could be used in more places? Yes, probably. For example, if there are linters that consume these constraints and which are fingerprinted, they should re-run when the constraints have changed... BUT, I think that those tasks are already dynamically selecting interpreters per-target, and thus aren't using this product.

All this to say: this feels like a good incremental step. Looking at the other `Task.invalidated` blocks for python tasks and seeing whether they're accounting for interpreters would be good. But I don't think it's a blocker here.